### ISSUE TITLE
The pulse of GitBlit

### ISSUE THREAD

**Issue by silvernode** [gitblit/gitblit#1274]
Just noticed how long ago the last commit was and it's getting close to 2 years since the last release. Is everything ok?

**Comment by rnveach** [comment]
Similar to https://github.com/gitblit/gitblit/issues/1265 .

**Comment by goettl79** [comment]
What about a fork? Seems to me there are some people interested...

**Comment by rnveach** [comment]
Just my opinion, but some of the problems I see with forking is 
* the use of this custom build, instead of using something more common like maven (https://github.com/gitblit/gitblit/issues/777)
* the use of old libraries which would probably take alot of work to upgrade, especially wicket (https://github.com/gitblit/gitblit/pull/1231)
* the lack of tests, and current tests that just fail (https://github.com/gitblit/gitblit/issues/1126)

Anyone who wants to take this on seriously would require alot of work.

**Comment by goettl79** [comment]
Well, these three issues seem to be kind of technical problems that could stop people from contributing.

However, I don't see how solving these issues  would solve the community problem of doing releases and accepting pull requests. Nobody is doing this.

The community seems to be active however. Issues are raised and pull requests created. Best would be new maintainers. However I am not sure if the original project initiators are still interested in the project..

**Comment by fzs** [comment]
I agree with rnveach and I would even go so far as to say that the lack of tests is certainly an issue that prevents accepting pull requests.

**Comment by pingunaut** [comment]
Hi @all, 

Issues like #1284 show that gitblit really still has a fan base and in my opinion, it has one advantage over all the big players like GitLab: It's small, simple and super easy to setup and use.
A lot of companies and developers have Issue Trackers, CI Platforms, Wikis, Code Review Tools, etc.  already in place.
For a number of people (me included), these features are not needed in a Git server. These people just need a small and simple Git server.

I fully agree with @rnveach and his three major pain points. However, I think it's possible fix them.
A 2.0.0 release moving to maven, having CI in place, building up some test coverage, upgrading libs and (also important in my opinion) updating to a more modern CSS layout would be a great win all gitblit users. This would also make it a lot easier to tackle existing and new issues.

The discussion here and also in #1265 shows that there are people willing to help out.
@fzs what do you think about creating a new 'dev' or '2.x' branch with access for some more people. This way the stability of the master won't decrease, while people who want to help, can do so more easily within the project. Releases from this branch could be marked as beta/unstable until some quality is ensured.
@goettl79 what's your opinion on that? Would that be an alternative for creating a fork?

Sorry for the long post. Would be cool to have some other platform than comment's on multiple issues  for a more detailed discussion.. Slack channel or something?

Regards,
Martin

**Comment by canol** [comment]
I just wanted to add that nowadays I was looking into some self-hosted project management options and Gitblit is still a very good one. It is simple, has almost all the basic features and its UI is pretty intuitive and elegant. The biggest disadvantage seems to be that the project looks abandoned...
